### ISSUE TITLE
"Warning: initial guess function not good" when using AD with a list_function as initial guess

### ISSUE THREAD

**Issue by Ferlemann** [Quantum-OCS/QuOCS#43]
In example "execute_AD.py", replacing the lines

```
if __name__ == "__main__":
    main(readjson(os.path.join(os.getcwd(), "settings_AD.json")))
```
by 
```
if __name__ == "__main__":
    settings_AD = readjson(os.path.join(os.getcwd(), "settings_AD.json"))
    settings_AD["pulses"][0]["initial_guess"] = {"function_type":  "list_function", "list_function": np.zeros(settings_AD["pulses"][0]["bins_number"])}
    main(settings_AD)
```

triggers the following message
```
> Warning: initial guess function not good. Do with 0.0
```

**Comment by Ferlemann** [comment]
Quick-fix by importing "jax.numpy" in "BasePulse.py":
```
import jax.numpy as jnp
```
and adjustment of the "_get_initial_guess" function:
```
    def _get_initial_guess(self) -> np.ndarray:
        """
        Gets the initial guess pulse

        :return np.ndarray: The initial guess pulse
        """
        if isinstance(self.initial_guess_pulse, Callable):
            initial_guess_t = self.initial_guess_pulse(self.time_grid)
        elif isinstance(self.initial_guess_pulse, np.ndarray):
            initial_guess_t = self.initial_guess_pulse
        elif isinstance(self.initial_guess_pulse, jnp.ndarray):
            initial_guess_t = self.initial_guess_pulse
        else:
            # TODO Handle here
            print("Warning: initial guess function not good. Do with 0.0")
            initial_guess_t = (lambda t: 0.0)(self.time_grid)
        return self._get_limited_pulse(initial_guess_t)
```

**Comment by ThomasReisser90** [comment]
Fixed in version 0.0.61
